python/ray/tests/test_task_metrics.py

```diff
@@ -598,6 +599,63 @@ def g():
         )
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="Flaky on Windows. Timing out.")
+def test_metrics_batch(shutdown_only):
```

Comment:
does this test fail before the patch?

Reply:
The point is when you batch, it ensures the corretness. So it doesn't fail before the patch. I checked the log that it sends 1 metrics in multiple grpc requests. If we want to test this part, we should write the whole unit test suite.. (I can do it if you think it is a blocker, but it is a lot of work)